fcrepo-fixity/src/test/java/org/fcrepo/camel/fixity/RouteTest.java

```diff
@@ -17,122 +17,127 @@
  */
 package org.fcrepo.camel.fixity;
 
-import static org.apache.camel.util.ObjectHelper.loadResourceAsStream;
-import static org.fcrepo.camel.FcrepoHeaders.FCREPO_URI;
-
-import java.util.Properties;
-
+import org.apache.camel.CamelContext;
 import org.apache.camel.EndpointInject;
 import org.apache.camel.Produce;
 import org.apache.camel.ProducerTemplate;
-import org.apache.camel.builder.AdviceWithRouteBuilder;
+import org.apache.camel.builder.AdviceWith;
 import org.apache.camel.component.mock.MockEndpoint;
-import org.apache.camel.test.blueprint.CamelBlueprintTestSupport;
+import org.apache.camel.model.ModelCamelContext;
+import org.apache.camel.spring.javaconfig.CamelConfiguration;
 import org.apache.commons.io.IOUtils;
-
+import org.junit.BeforeClass;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.AnnotationConfigContextLoader;
+
+import static org.apache.camel.component.mock.MockEndpoint.assertIsSatisfied;
+import static org.apache.camel.util.ObjectHelper.loadResourceAsStream;
+import static org.fcrepo.camel.FcrepoHeaders.FCREPO_URI;
 
 /**
  * Test the route workflow.
  *
  * @author Aaron Coburn
  * @since 2015-06-18
  */
-public class RouteTest extends CamelBlueprintTestSupport {
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = {RouteTest.ContextConfig.class}, loader = AnnotationConfigContextLoader.class)
+public class RouteTest {
 
-    @EndpointInject(uri = "mock:result")
+    private static final long ASSERT_PERIOD_MS = 5000;
+    @EndpointInject("mock:result")
     protected MockEndpoint resultEndpoint;
 
-    @Produce(uri = "direct:start")
+    @Produce("direct:start")
     protected ProducerTemplate template;
 
     private static final String baseURL = "http://localhost/rest";
     private static final String identifier = "/file1";
 
-    @Override
-    public boolean isUseAdviceWith() {
-        return true;
-    }
+    @Autowired
+    private CamelContext camelContext;
 
-    @Override
-    public boolean isUseRouteBuilder() {
-        return false;
-    }
+    @Autowired
+    private FcrepoFixityConfig config;
 
-    @Override
-    protected String getBlueprintDescriptor() {
-        return "/OSGI-INF/blueprint/blueprint-test.xml";
-    }
+    @BeforeClass
+    public static void beforeClass() {
+        System.setProperty("fixity.failure", "mock:failure");
+        System.setProperty("fixity.success", "mock:success");
+        System.setProperty("fixity.input.stream", "seda:foo");
+        System.setProperty("fixity.enabled", "true");
```

Comment:
Would it make sense to have a test where enabled is false to show that it works? But only if its practical to do so.

Reply:
I'll look into it.

Reply:
Added.  I tried to it using @BeforeTestExecution or @BeforeTestMethod but those weren't being triggered probably because I'm not using a spring test base class.  Anyway, I resolved by adding a separate test class.